### PR TITLE
DTPORTAL-19293 SF Agent Panel (Classic) - Login fails in Chrome

### PR DESCRIPTION
**Pull request by abhishek-algoworks**
### Code Reviewers
@BMTTMC @tdupreeibp @gfaza

### CODE REVIEW

File: library/Zend/Session.php [hunk 1]
```diff
@@ -321,6 +321,12 @@ public static function regenerateId()
         } else {
             if (!self::$_unitTestEnabled) {
                 session_regenerate_id(true);
+
+                // This can be removed once we update to 7.4 or above as then
+                // we can use php.ini directive to achieve the same `session.cookie_samesite=Lax`
+                if (version_compare(PHP_VERSION, '7.3.0', '<')) {
```
Comment (BMTTMC) [inline library/Zend/Session.php]: Even once we do upgrade to 7.4, I think we need to keep this line in util we add the PHP directive. Therefore, it seems to me should always run this, independent of the PHP version.

File: library/Zend/Session.php [hunk 2]
```diff
@@ -321,6 +321,12 @@ public static function regenerateId()
         } else {
             if (!self::$_unitTestEnabled) {
                 session_regenerate_id(true);
+
+                // This can be removed once we update to 7.4 or above as then
+                // we can use php.ini directive to achieve the same `session.cookie_samesite=Lax`
```
Comment (BMTTMC) [inline library/Zend/Session.php]: I think this comment should specify session.cookie_samesite=None.

File: library/Zend/Session.php (hunk 1 above)
Comment (abhishek-algoworks) [reply library/Zend/Session.php]: This change will throw an error for 7.3 and above. There is an alternate code we can add for 7.3 and above but I did not want to add anything that cannot be tested. We can add the below code in an else statement:
```
setcookie('IFBYPHONE', self::getId(), [
    'expires' => 0,
    'path' => '/',
    'secure' => true,
    'httponly' => true,
    'samesite' => 'None',
]);```

File: library/Zend/Session.php (hunk 2 above)
Comment (abhishek-algoworks) [reply library/Zend/Session.php]: Fixed.

**Comment by BMTTMC** [comment]
I think we do need to add support for 7.4, so that we don't need to coordinate the upgrade with this specific code change.  I would vote for putting in the code for 7.3+, but I am interested in what Tim and Frederic think.

**Comment by tdupreeibp** [comment]
I would like to see an else that handles 7.4.  We need to make sure we come back to clean this up, but I don't want this to automatically make a 7.4 implementation fail.

**Comment by abhishek-algoworks** [comment]
@BMTTMC @tdupreeibp Updated to handle 7.3+ update.

**Comment by BMTTMC** [comment]
Thank you for making this change and adding 7.4 support.  It looks good to my untrained eye, but Tim knows this code much better than I do, so I'll hold off final approval until he's had a chance to review.

**Comment by BMTTMC** [comment]
I noticed that the unit tests are failing.  It doesn't look like it's related to this change; do you know the cause?

**Comment by tdupreeibp** [comment]
@BMTTMC @abhishek-algoworks Looks good, thank you

**Comment by tdupreeibp** [comment]
@BMTTMC in regards to the failed build, this is what I see in travis
Screen Shot 2020-09-16 at 2 12 04 PM

**Comment by gfaza** [comment]
I'm pretty sure these were failing in the zf1-future from last year, as well.  They were dismissed at the time for the sake of expedience.  I took 30 min to try and resolve them, but without much luck (I only managed to kick the errors further down the build).  I don't have a ton of bandwidth or Travis expertise to bring to the table, but am interested to see if these builds can get to the green in a cost-effective manner.

**Comment by tdupreeibp** [comment]
@BMTTMC @abhishek-algoworks @gfaza are we good to go on this then?

**Comment by gfaza** [comment]
@tdupreeibp, this looks credible to me especially if it works!  my only comment would be, that it looks like we might be able to DRY it up a little.

@abhishek-algoworks, it looks like this is the same code in two places.  could this code be factored out to something like the following?
```
/**
 * @todo Remove definition and calls after upgrade to PHP 7.4
 *
 * This can be removed once we update to 7.4 or above as then
 * we can use php.ini directive to achieve the same `session.cookie_samesite=None`
 */
public static function sameSiteCookieWorkaroundPrePhp74()
{
    if (version_compare(PHP_VERSION, '7.3.0', '<')) {
        setcookie('IFBYPHONE', self::getId(), 0, '/; SameSite=None; HttpOnly; Secure');
    } else {
        setcookie('IFBYPHONE', self::getId(), [
            'expires'  => 0,
            'path'     => '/',
            'secure'   => true,
            'httponly' => true,
            'samesite' => 'None'
        ]);
    }
}
```

**Comment by abhishek-algoworks** [comment]
@gfaza Done. Please review.

**Comment by gfaza** [comment]
👍  lgtm!

**Review by BMTTMC — APPROVED** [review]
Changes look good.  Ship it! :shipit: 👍

**Review by jbuehring — APPROVED** [review]
:crown:

**Pull request merged**